pkg/cmd/get/clusters.go

```diff
@@ -74,9 +74,27 @@ func list(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	} else {
-		// Convert the list of the clusters to Table of clusters
-		printTable(printers.PrintOptions{}, generateClusterTable(clusters))
-		return nil
+		// Convert the list of the clusters to a Table of clusters and print the table using the format selected
+		err := printClustersOutput(os.Stdout, clusters, outputFormat)
+		if err != nil {
+			return err
+		} else {
+			return nil
+		}
+	}
+}
+
+func printClustersOutput(outWriter io.Writer, clusters []Cluster, format string) error {
+	switch format {
+	case "json":
+		return util.PrintDataAsJson(clusters, outWriter)
+	case "yaml":
+		return util.PrintDataAsYaml(clusters, outWriter)
+	case "":
+		return util.PrintTable(generateClusterTable(clusters), outWriter)
+	default:
+
+		return fmt.Errorf("output format %s is not supported", format)
```

Comment:
Can we combine this function for secret and cluster both and add it in utils package to avoid duplicate code?

Reply:
> Can we combine this function for secret and cluster both and add it in utils package to avoid duplicate code?

We can do that but then there is still an issue as the Table to be generated will change according to the type: Secret, Cluster, Package and then even if the code is generic, then it will be needed to have a switch or if then statement to figure out if we generate the Table for a Secret, Cluster or Package within the package printers.

**Note**: Ideally the Table to be printed should be generated within the code of the package secret, clusters, etc

Dummy example
```
func printOutput[T any](outWriter io.Writer, input []T, format string) error {
	switch format {
	case "json":
		return printer.PrintDataAsJson(input, outWriter)
	case "yaml":
		return printer.PrintDataAsYaml(input, outWriter)
	case "table":
		// If item type of []any is Secret then we call
		return printer.PrintTable(generateClusterTable(results), outWriter)
		// If item type of []any is Secret then we call
		return printer.PrintTable(generateSecretTable(results), outWriter)
	default:
		return fmt.Errorf("output format %s is not supported", format)
	}
}
```
Idea are welcome ;-) @punkwalker